client/python/gradio_client/utils.py

```diff
@@ -387,6 +387,16 @@ def encode_url_or_file_to_base64(path: str | Path):
     return encode_file_to_base64(path)
 
 
+def download_byte_stream(url: str, hf_token=None):
+    arr = bytearray()
+    headers = {"Authorization": "Bearer " + hf_token} if hf_token else {}
+    with httpx.stream("GET", url, headers=headers) as r:
+        for data in r.iter_bytes():
+            arr += data
+            yield data
+    yield arr
```

Comment:
Am I understanding correctly that what's happening is that we are iterating chunk by chunk and then finally yielding the full array so that `.predict()` returns the full data array? 2 questions:

(1) How come outputs() doesn't include this final full data array?
(2) Will `r.iter_bytes()` work if the streaming response returns binary data (as in #5238) instead of file data?

Reply:
(1) outputs only includes the values returned from the server during the `process_generating` event
(2) yea the `stream` enpoint only serves bytes so it should work when #5238 is merged